### ISSUE TITLE
Linking to ID's works only some of the times.

### ISSUE THREAD

**Issue by pjeurien** [naturalcrit/homebrewery#3272]
### Renderer

v3

### Browser

Chrome

### Operating System

Windows

### What happened?

I am creating a list of links in a markdown table and it does not work for all off the links.

None of the links with an & works, they get stripped out in the id. 
Eg: meats & more becomes meats--more.

For others it's a mixed bag.  Sometimes they work, sometimes they don't. The id's are not duplicated in the brew. 

### Code

```gfm
|Type|Shop|District|
|:----|:----|:----|
|Antiquities|[Relic’s & Rarities](#relics--rarities)|Veiled Corner|
|Bakery|[Oliver’s Bakery](#olivers-bakery)|Park Way|
|Blacksmith|[Sutah’s Flame](#sutahs-flame)|The Commons|
|Bookstore|[Badai’s Books and Bindings](#badais-books-and-bindings)|Elder’s Seat|
|Butcher|[Meats & More](#meats--more)|The Commons|
|Conservatorium|[Baroc’s Conservatorium](#barocs-conservatorium)|Artist’s Quarter|
|Flowershop|[The Pettle Trail](#the-pettle-trail)|Park Way|
|General Store|[Roman’s General Store](#romans-general-store)|The Commons|
|Inn|[The Little Lantern Lounge](#the-little-lantarn-lounge)|-|
|Inn|The Old Haunt|Park Way|
|Stable|[Pets & Paws](#pets--paws)|Park Way|
|Tabacco|Tabar’s Tabacco|The Commons|
|Tailor|[Adeline’s Wardrobe](#adelines-wardrobe)|Artist’s Quarter|
|Temple|Church of the Fey|Mourner’s Row|
```

**Comment by 5e-Cleric** [comment]
Here's the thing, for now, if a brew is longer than 50 pages, Partial Page Rendering is activated, which only renders a few pages to improve performance.

When that happens, a link will only work between rendered pages, as reported in #1923, recently fixed.
But that will only fix page links, which we use in the ToC, for the rest of links we are working on a fix, #3262, which should keep a record of all headings so you can link to them even if they are not rendered.

TLDR: we're already on it.

**Comment by dbolack-ab** [comment]
Regarding the Ampersands, the Heading code uses the GHM slugging rules to generate the id which undoubtedly removes characters that have meanings in URLs.

**Comment by dbolack-ab** [comment]
Are there still outstanding problems with this issue?

**Comment by 5e-Cleric** [comment]
https://github.com/user-attachments/assets/5498cf73-0f85-42d9-9db3-f6cafa3ea326

For some reason the heading's id increases with every save, making the id useless.

**Comment by 5e-Cleric** [comment]
Whatever this bug is, its juicy, i've managed to reproduce it with the following as minimal code:

```md


\page


\page
## test text


```

At least 2 pages are required, for some reason, and it is still not showing always, but it is definitely there

https://github.com/user-attachments/assets/6a76ccda-3a69-469b-8eb6-cf820e20682a

adding a more clean video
